WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,31 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    # Remove extra rounds.
+    self.rounds.each do |round|
+      wcif_round = wcif["rounds"][round.number - 1]
```

Comment:
Giving up the rich shapes JSON lets us have in favor of a more relational model is a change @keemy and I struggled with when working on CCM years ago (we were using mongodb, so we thought each competition could simply be one document in a competitions collection). We eventually switched to a more relational model, as you suggest here. However, what makes sense for a mongodb schema does not necessarily make sense for a JSON data format.

I do wonder if a more flat structure with unique ids would allow us to define "merging" WCIF documents in a useful way (something that came up when we were trying to write a "set wcif" api).

I don't think it's useful to distinguish between "static' and "dynamic" data with the WCIF. Everything can change, just some stuff changes more quickly than others.

Without a more compelling reason than "it seems cleaner", I think we should move forward with the currently documented WCIF (although I love the naming improvements we've come up with in the last week).